src/main/riddl/Events/eventContext.riddl

```diff
@@ -2,195 +2,666 @@ type EventId is Id (ImprovingApp.Events.EventContext.Event)
   briefly "Unique identifier for Tenant Entity"
 
 context EventContext is {
-    include "eventMessages"
-    include "eventTypeDefinitions"
-    include "eventSources"
+  include "eventMessages"
+  include "eventTypeDefinitions"
+  include "eventSources"
 
-    entity Event is {
-        option is aggregate
+  function incrementDelayedEventTime is {
+    requires {originalTime: DateTime, duration: Duration}
+    returns { newTime: DateTime }
+  } briefly "used for incrementing start time and end time when DelayEvent is received"
 
-        handler EventHandler is {
-            on command ScheduleEvent {
-                then morph entity Event to state Scheduled with
-                  !Event.ScheduledState(
-                    eventInfo = !EventInfo(),
-                    eventMetaInfo = !EventMetaInfo(),
-                    reservation = !ReservationId()
-                  )
-            }
-            on other {
-            then error "You must first create an event using ScheduleEvent command."
-            }
-        }
+  function startTimeIfPast is {
+    requires {startTime: DateTime}
+    returns { timeOpt: DateTime? }
+  } briefly "returns startTime if it is past, empty option if it isn't. This is used when CancelEvent is received"
+
+  function nowIfStartIsPast is {
+    requires {startTime: DateTime}
+    returns { nowOpt: DateTime? }
+  } briefly "returns now if startTime is past, empty option if it isn't. This is used when CancelEvent is received"
+
+  function newEditableInfoFromFieldsPresentInEditable is {
+    requires { editable: EditableEventInfo, oldInfo: EditableEventInfo }
+    returns { newInfo: EditableEventInfo }
+  } briefly "returns newInfo with all fields present in editable and all those not present there taken from oldInfo"
+
+  function newInfoFromFieldsPresentInEditable is {
+    requires { editable: EditableEventInfo, oldInfo: EventInfo }
+    returns { newInfo: EventInfo }
+  } briefly "returns newInfo with all fields present in editable and all those not present there taken from oldInfo"
+
+  entity Event is {
+    option is aggregate
+
+    handler EventHandler is {
+      on command CreateEvent {
+        then send event EventCreated(
+            eventId = @CreateEvent.eventId,
+            info = @CreateEvent.info,
+            meta = !EventMetaInfo(
```

Comment:
Looks like new RIDDL syntax that I need to get caught up on. What does !EventMetaInfo do? And how is that different than @CreateEvent.info (other than the specific reference details)?

Reply:
`@` references an object in-memory (usually a command or state)
`!` creates a new object